File: 0762-prime-number-of-set-bits-in-binary-representation/0762-prime-number-of-set-bits-in-binary-representation.py
class Solution:
    def countPrimeSetBits(self, left: int, right: int) -> int:
        primes = {2, 3, 5, 7, 11, 13, 17, 19}
        return sum(bin(val).count('1') in primes \
                   for val in range(left, right + 1))
        
        

File: 0762-prime-number-of-set-bits-in-binary-representation/0762-prime-number-of-set-bits-in-binary-representation.go
package main

import (
    "math"
)

func countPrimeSetBits(left int, right int) int {
    count := 0

    for n := left; n <= right; n++ {
        bits := countBits(n)
        if isPrime(bits) {
            count++
        }
    }

    return count
}

func countBits(n int) int {
    bits := 0
    for n > 0 {
        bits += n & 1
        n >>= 1
    }
    return bits
}

func isPrime(n int) bool {
    if n < 2 {
        return false
    }

    limit := int(math.Sqrt(float64(n)))
    for i := 2; i <= limit; i++ {
        if n%i == 0 {
            return false
        }
    }
    return true
}